File: Benchmarking/src/gen593.swift
  import Dispatch
  struct s0 {
    var p0: [[Double]]
    var p1: [[Double]]
    var p2: [[Double]]
    var p3: Double
    var p4: [[Double]]
  }
  struct s2 {
    var p0: s0
    var p1: [s0]
  }
  func f49(_ v0: [s2], _ v1: s2) -> s2 {
    var v4: [s2] = v0
    var v2: [s2] = v0
    let v11: s2 = v2[0]
    let v3: [s0] = v1.p1
    var v5: s2 = v1
    let v7: [s0] = v1.p1
    let v6: s0 = v7[0]
    let v24: s0 = v7[0]
    let v13: [s0] = v5.p1
    v5.p1 = v13
    let v10: [s0] = v1.p1
    var v18: s0 = v6
    let v12: s0 = v7[0]
    let v33: s0 = v5.p0
    let v14: Double = v6.p3
    v18.p3 = v14
    var v29: Double = v14
    v5.p1 = v10
    var v15: [s0] = v13
    var v49: s2 = v11
    var v21: Double = v14
    let v30: [Double] = [v21, v21, v14, v14, v29, v29, v14]
    var v71: s0 = v18
    let v39: [[Double]] = v12.p4
    let v35: s0 = v49.p0
    v4[1] = v5
    var v41: [s0] = v3
    var v52: [s0] = v41
    let v86: Double = v30[0]
    v49.p1 = v41
    let v19: s0 = v52[1]
    let v31: Double = v21 * v21
    v2[0] = v1
    v15[1] = v71
    let v28: s2 = v4[0]
    v5.p0 = v6
    var v20: [s0] = v3
    v21 = v31
    v71.p0 = v39
    v2[1] = v49
    v15 = v20
    var v40: Double = v86
    var v53: [Double] = v30
    v71.p3 = v86
    var v102: s2 = v5
    let v89: s0 = v13[0]
    var v51: [Double] = v53
    v102.p0 = v24
    let v109: Double = v40 / v31
    v5.p0 = v6
    let v116: s0 = v10[0]
    let v47: [s2] = [v49, v5, v11, v49, v102, v28, v1]
    v18.p4 = v39
    v49.p0 = v89
    var v42: [s2] = v47
    v102.p0 = v33
    v53[1] = v40
    v102.p1 = v41
    v102.p0 = v19
    v49.p1 = v15
    let v103: Double = v51[5]
    v71.p3 = v103
    v5.p0 = v116
    v102.p1 = v52
    v5.p0 = v12
    v52[0] = v89
    let v88: [[Double]] = v35.p0
    v18.p0 = v88
    v5.p0 = v89
    let v97: s2 = v42[5]
    v4[1] = v49
    v18.p3 = v109
    v5.p1 = v10
    v53[6] = v103
    return v97
  }
  @inline(never)
  func f0(_ v0: [s2], _ v1: Double) -> Double {
    let v6: s2 = v0[0]
    let v3: s2 = f49(v0, v6)
    let v9: s0 = v3.p0
    let v11: [[Double]] = v9.p2
    let v61: [Double] = v11[1]
    let v66: Double = v61[0]
    return v66
  }
  func benchmark() {
    let v0: [s2] = [s2(p0: s0(p0: [[0.0]], p1: [[1.0], [2.0], [3.0]], p2: [[4.0], [5.0], [6.0]], p3: 7.0, p4: [[8.0]]), p1: [s0(p0: [[9.0]], p1: [[10.0], [11.0], [12.0]], p2: [[13.0], [14.0], [15.0]], p3: 16.0, p4: [[17.0]]), s0(p0: [[18.0]], p1: [[19.0], [20.0], [21.0]], p2: [[22.0], [23.0], [24.0]], p3: 25.0, p4: [[26.0]])]), s2(p0: s0(p0: [[27.0]], p1: [[28.0], [29.0], [30.0]], p2: [[31.0], [32.0], [33.0]], p3: 34.0, p4: [[35.0]]), p1: [s0(p0: [[36.0]], p1: [[37.0], [38.0], [39.0]], p2: [[40.0], [41.0], [42.0]], p3: 43.0, p4: [[44.0]]), s0(p0: [[45.0]], p1: [[46.0], [47.0], [48.0]], p2: [[49.0], [50.0], [51.0]], p3: 52.0, p4: [[53.0]])])]
    let v1: Double = 54.0
    let start = DispatchTime.now().uptimeNanoseconds
    var result: Double = 54.0
    for _ in 1...1000 {
      result = f0(v0, v1)
    }
    let end = DispatchTime.now().uptimeNanoseconds
    print(result)
    print(end - start)
  }
  benchmark()

File: Benchmarking/src/gen593.cpp
  #include <vector>
  #include <iostream>
  #include <chrono>
  struct s0 {
    std::vector<std::vector<double>> p0;
    std::vector<std::vector<double>> p1;
    std::vector<std::vector<double>> p2;
    double p3;
    std::vector<std::vector<double>> p4;
    s0(std::vector<std::vector<double>> v0, std::vector<std::vector<double>> v1, std::vector<std::vector<double>> v2, double v3, std::vector<std::vector<double>> v4): p0(v0), p1(v1), p2(v2), p3(v3), p4(v4) { }
  };
  struct s2 {
    s0 p0;
    std::vector<s0> p1;
    s2(s0 v0, std::vector<s0> v1): p0(v0), p1(v1) { }
  };
  s2 f49(const std::vector<s2> &v0, const s2 &v1) {
    std::vector<s2> v4 = v0;
    std::vector<s2> v2 = v0;
    const s2 v11 = v2[0];
    const std::vector<s0> v3 = v1.p1;
    s2 v5 = v1;
    const std::vector<s0> v7 = v1.p1;
    const s0 v6 = v7[0];
    const s0 v24 = v7[0];
    const std::vector<s0> v13 = v5.p1;
    v5.p1 = v13;
    const std::vector<s0> v10 = v1.p1;
    s0 v18 = v6;
    const s0 v12 = v7[0];
    const s0 v33 = v5.p0;
    const double v14 = v6.p3;
    v18.p3 = v14;
    double v29 = v14;
    v5.p1 = v10;
    std::vector<s0> v15 = v13;
    s2 v49 = v11;
    double v21 = v14;
    const std::vector<double> v30 { v21, v21, v14, v14, v29, v29, v14 };
    s0 v71 = v18;
    const std::vector<std::vector<double>> v39 = v12.p4;
    const s0 v35 = v49.p0;
    v4[1] = v5;
    std::vector<s0> v41 = v3;
    std::vector<s0> v52 = v41;
    const double v86 = v30[0];
    v49.p1 = v41;
    const s0 v19 = v52[1];
    const double v31 = v21 * v21;
    v2[0] = v1;
    v15[1] = v71;
    const s2 v28 = v4[0];
    v5.p0 = v6;
    std::vector<s0> v20 = v3;
    v21 = v31;
    v71.p0 = v39;
    v2[1] = v49;
    v15 = v20;
    double v40 = v86;
    std::vector<double> v53 = v30;
    v71.p3 = v86;
    s2 v102 = v5;
    const s0 v89 = v13[0];
    std::vector<double> v51 = v53;
    v102.p0 = v24;
    const double v109 = v40 / v31;
    v5.p0 = v6;
    const s0 v116 = v10[0];
    const std::vector<s2> v47 { v49, v5, v11, v49, v102, v28, v1 };
    v18.p4 = v39;
    v49.p0 = v89;
    std::vector<s2> v42 = v47;
    v102.p0 = v33;
    v53[1] = v40;
    v102.p1 = v41;
    v102.p0 = v19;
    v49.p1 = v15;
    const double v103 = v51[5];
    v71.p3 = v103;
    v5.p0 = v116;
    v102.p1 = v52;
    v5.p0 = v12;
    v52[0] = v89;
    const std::vector<std::vector<double>> v88 = v35.p0;
    v18.p0 = v88;
    v5.p0 = v89;
    const s2 v97 = v42[5];
    v4[1] = v49;
    v18.p3 = v109;
    v5.p1 = v10;
    v53[6] = v103;
    return v97;
  }
  __attribute__((noinline))
  double f0(const std::vector<s2> &v0, const double &v1) {
    const s2 v6 = v0[0];
    const s2 v3 = f49(v0, v6);
    const s0 v9 = v3.p0;
    const std::vector<std::vector<double>> v11 = v9.p2;
    const std::vector<double> v61 = v11[1];
    const double v66 = v61[0];
    return v66;
  }
  int main() {
    std::vector<s2> v0({ { { { { 0.0 } }, { { 1.0 }, { 2.0 }, { 3.0 } }, { { 4.0 }, { 5.0 }, { 6.0 } }, 7.0, { { 8.0 } } }, { { { { 9.0 } }, { { 10.0 }, { 11.0 }, { 12.0 } }, { { 13.0 }, { 14.0 }, { 15.0 } }, 16.0, { { 17.0 } } }, { { { 18.0 } }, { { 19.0 }, { 20.0 }, { 21.0 } }, { { 22.0 }, { 23.0 }, { 24.0 } }, 25.0, { { 26.0 } } } } }, { { { { 27.0 } }, { { 28.0 }, { 29.0 }, { 30.0 } }, { { 31.0 }, { 32.0 }, { 33.0 } }, 34.0, { { 35.0 } } }, { { { { 36.0 } }, { { 37.0 }, { 38.0 }, { 39.0 } }, { { 40.0 }, { 41.0 }, { 42.0 } }, 43.0, { { 44.0 } } }, { { { 45.0 } }, { { 46.0 }, { 47.0 }, { 48.0 } }, { { 49.0 }, { 50.0 }, { 51.0 } }, 52.0, { { 53.0 } } } } } });
    double v1(54.0);
    auto start = std::chrono::high_resolution_clock::now();
    double result;
    for (int i = 0; i < 1000; i ++) {
      result = f0(v0, v1);
    }
    auto end = std::chrono::high_resolution_clock::now();
    double value = *((double*) &result);
    std::cout << value << "\n";
    std::cout << std::chrono::duration_cast<std::chrono::nanoseconds>(end-start).count();
    std::cout << "\n";
    return 0;
  }
